src/assertDeepEqual.lua

```diff
@@ -19,7 +19,7 @@ local function deepEqual(a, b)
 			visitedKeys[key] = true
 
 			local success, innerMessage = deepEqual(value, b[key])
-			if not success then
+			if not success and innerMessage then
```

Comment:
Unless there's a narrowing bug lingering, using the above type suggestion should eliminate the needs for these extra checks. If it doesn't, add a `-- FIXME Luau` comment and we can help file a Luau issue.